src/main/java/com/amazon/opendistroforelasticsearch/security/rest/TenantInfoAction.java

```diff
@@ -137,7 +144,37 @@ public void accept(RestChannel channel) throws Exception {
             }
         };
     }
-    
+
+    private boolean isUnAuthorized() {
+        final User user = (User)threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+
+        // check if the user is a kibanauser or super admin
+        boolean initialCheck =   user == null ||
+                (!user.getName().equals(evaluator.kibanaServerUsername()))
+                        && !adminDns.isAdmin(user);
+        if(!initialCheck){
+            return false;
+        }
+
+        // check if the roles belong to kibana opendistro role
+        final SecurityDynamicConfiguration<?> configuration = load(CType.ROLESMAPPING, true);
+
+        String kibanaOpendistroRole = evaluator.kibanaOpendistroRole();
+
+        if (configuration.exists(kibanaOpendistroRole)) {
+            RoleMappings rolesMapping = (RoleMappings) configuration.getCEntries().get(kibanaOpendistroRole);
+            if(rolesMapping.getUsers().contains(user.getName())){
```

Comment:
Doesnt the privileges evaluator already check for role mappings for the user ? 
https://github.com/opendistro-for-elasticsearch/security/blob/master/securityconfig/roles_mapping.yml#L46

Reply:
Also the rolemappings can be done via opendistro security roles inside an internal users so check that as well?

Reply:
> opendistro security roles inside an internal us

We cant use this as this is relevant from es 7.4 and above. 
Below 7.4 doesnt have the same.

Reply:
> Doesnt the privileges evaluator already check for role mappings for the user ?
> https://github.com/opendistro-for-elasticsearch/security/blob/master/securityconfig/roles_mapping.yml#L46

It doesnt look like, it has any method or config which will give me role mappings for the user